projects/ngx-darkbox-gallery/src/lib/components/darkbox/darkbox.component.ts

```diff
@@ -50,9 +51,15 @@ export class DarkboxComponent implements OnInit {
    */
   fullSizedImageLoaded = false;
 
-  constructor() { }
+  /**
+   * Interpolated image caption
+   */
+  imageCaption: string;
+
+  constructor(private imageCaptionService: ImageCaptionService) { }
 
   ngOnInit(): void {
+    this.imageCaption = this.imageCaptionService.getImageCaption(this.darkboxConfiguration, this.currentNumber, this.totalNumber, this.image.caption);
```

Comment:
How does the caption change onNext and onPrev? Don't we have to define a function and call it everywhere, the currently displayed image is updated?

Reply:
I used the angular update cycle now. Please check if it works correctly now.